frontends/p4-14/typecheck.cpp

```diff
@@ -16,15 +16,19 @@ limitations under the License.
 
 #include "typecheck.h"
 
-// P4 v1.0 and v1.1 type checking algorithm
-// Initial type setting based on immediate context:
-// - replace named reference to ActionParams in the bodies of ActionFunctions with the
-//   actual ActionParam
-// - link header/metadata instances to the Type
-// - replace named references to global header or metadata instances with ConcreteHeaderRef
-//   expressions that link directly to them.
-// - set type for Member and HeaderStackItemRefs
+/// P4-14 (v1.0 and v1.1) type checking algorithm
+/// Initial type setting based on immediate context:
+/// - replace named reference to ActionParams in the bodies of ActionFunctions with the
+///   actual ActionParam
+/// - link header/metadata instances to the Type
+/// - replace named references to global header or metadata instances with ConcreteHeaderRef
+///   expressions that link directly to them.
+/// - set type for Member and HeaderStackItemRefs
 class TypeCheck::AssignInitialTypes : public Transform {
```

Comment:
I generally define nested visitors for a PassManager subclass as nested classes in the PassManager subclass; seems cleaner to me that defining a separate namespace in addition to the class.